src/twisted/python/_setup.py

```diff
@@ -238,6 +238,7 @@ class my_build_ext(build_ext_twisted):
         "incremental >= 16.10.1",
         "Automat >= 0.3.0",
         "hyperlink >= 17.1.1",
+        "PyHamcrest >= 1.9.0",
```

Comment:
Because the tests are packaged under `twisted` with the implication that users can run at least a subset of them in a fresh installation.

If I put `pyhamcrest` in the `dev` extra's requirements, then I would have had to put a skip on the tests that use it and a conditional import.  There are enough tests that I think would benefit from `pyhamcrest` that making it a `dev`-only dependency would result in many additional skips. 

It's also a small, pure-Python dependency that's only used in private test modules, so including a dependency on it seemed low risk.